server/service/datagram/push.go

```diff
@@ -0,0 +1,81 @@
+// Package datagram defines all actions
+// associated with the GRPC service "Datagram".
+package datagram
+
+import (
+	"context"
+	"database/sql"
+
+	pb "github.com/UCSC-CSE123/beavertail/server/beavertail"
+)
+
+// Server is an unimplemented push datagram server.
+type Server struct {
+	db *sql.DB
+}
+
+// NewServer takes in all the required dependencies and returns
+// a new datagram server.
+func NewServer(db *sql.DB) *Server {
+	return &Server{
+		db: db,
+	}
+}
+
+func (srv *Server) insertIntoDB(req *pb.DatagramPush) error {
+	insertStatement := `
+	INSERT INTO Passengers (id, count, confidence, time)
+	VALUES ('$1', '$2', '$3', '$4');`
+
+	_, err := srv.db.Exec(insertStatement,
+		req.GetBusID(),
+		req.GetPassengerCount(),
+		req.GetPassengerCountConfidence(),
+		req.GetTimestamp(),
+	)
+
+	return err
+}
+
+// Push implements the protocol buffer definition.
+func (srv *Server) Push(ctx context.Context, req *pb.DatagramPush) (*pb.DatagramAck, error) {
+
+	// TODO: Implement context handling.
+
+	err := ctx.Err()
+	if err != nil {
+		return &pb.DatagramAck{
+			Acknowledgment: pb.DatagramAck_BUSY,
+		}, err
+	}
+
+	// Insert into the DB.
+	if err := srv.insertIntoDB(req); err != nil {
+		// If there was an error it doesn't necessarily mean
+		// it was a bad query.
+		//
+		// We'll send a busy signal for now.
+		// TODO: Find out how errors are handled by GRPC.
+		return &pb.DatagramAck{
+			Acknowledgment: pb.DatagramAck_BUSY,
+		}, err
+
+	}
+	// If there was no error,
+	// return a point to a datagram ack.
+	return &pb.DatagramAck{
+		Acknowledgment: pb.DatagramAck_OK,
+	}, nil
+}
+
+// InitDatabase initializes the database associated with this server.
+func (srv *Server) InitDatabase() error {
```

Comment:
Why is this here and not in `server/main.go`? Also, is there a reason that `InitDatabase` and `NewDB` are separate?

Reply:
I'd like to keep `server/main.go` as lean as possible and try to have handle only the command line items there. `NewDB` is seperate from `InitDatabase` because `NewDB` just opens a new connection to the database and make sures its okay, whereas `InitDatabase` specifically inits the database for our datagram service, thus the reason it's in the `server/service/datagram`. 

I'm happy to make any changes though (:

Reply:
For sure. Intuitively, I don't see a reason for them to be separated, since both functions are called unconditionally in `server/main.go` and they are both idempotent.

Reply:
After more thought I think your reasoning is correct, because if we do add more servers hypothetically, they would all be initing the database and I'm pretty sure thats no bueno. I'll seperate these and merge (: